core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -186,4 +194,19 @@ private static String scheme(String location) {
 
     return null;
   }
+
+  @SuppressWarnings("checkstyle:NoFinalizer")
+  @Override
+  protected void finalize() throws Throwable {
```

Comment:
Sorry @nastra, but stong -100 on using `finalize()` at all. See [JDK-8165641](https://bugs.openjdk.org/browse/JDK-8165641].

Reply:
@snazy, we know.

Iceberg uses this pattern to ensure that close is called. We also log the stack trace at WARN so that we can track down leaks. It is better to close in a finalizer than to not close at all.

Reply:
@snazy I agree that `finalize()` is a pattern we shouldn't rely on, but on Spark executor notes we don't have any other way of closing resources unfortunately